index.js

```diff
@@ -24,8 +25,8 @@ module.exports = postcss.plugin('postcss-remove-unused', ({html, preserveFlags =
 					return;
 				}
 
-				if (node.selector && !node.selector.match(/:(?:not)/)) {
-					let selector = node.selector.replace(/::?[\w-]+/g, '');
+				if (node.selector && !/(?:^|[,])\s*:not/.test(node.selector)) {
```

Comment:
this regex is a little odd, what's it supposed to match? what's the `^` for and why is the `,` in a character group?

Reply:
The character group is not necessary, I think I had multiple characters at one point. I've removed it. I also added some comments and examples for the regular expressions.

Reply:
Hang on a sec, I lied in one of my comments

Reply:
Ok, it should be good now. I just realized that selectors like `:not(.foo) .bar` can be safely removed if `.bar` has no matches.